app/v2/controllers/actions/MessageSizeAction.scala

```diff
@@ -31,23 +31,23 @@ import scala.concurrent.ExecutionContext
 import scala.concurrent.Future
 import scala.util.Try
 
-class MessageSizeAction @Inject() (config: AppConfig)(implicit val executionContext: ExecutionContext) extends ActionRefiner[Request, Request] {
+class MessageSizeAction[R[_] <: Request[_]] @Inject() (config: AppConfig)(implicit val executionContext: ExecutionContext) extends ActionRefiner[R, R] {
 
-  override protected def refine[A](request: Request[A]): Future[Either[Result, Request[A]]] =
+  override protected def refine[A](request: R[A]): Future[Either[Result, R[A]]] =
     contentLengthHeader(request) match {
       case Some((_, length)) => Future.successful(checkSize(length, request))
       case None =>
         Future.successful(Left(BadRequest(Json.toJson(BaseError.badRequestError("Missing content-length header"))))) // should never happen
     }
 
-  private def checkSize[A](length: String, request: Request[A]): Either[Result, Request[A]] =
+  private def checkSize[A](length: String, request: R[A]): Either[Result, R[A]] =
```

Comment:
I must have missed this before but the toOption on the Try isn't needed. You can pattern match on Success and Failure

`  private def checkSize[A](length: String, request: R[A]): Either[Result, R[A]] =
    Try(length.toInt) match {
      case Success(x) if x <= limit => Right(request)
      case Success(_)               => Left(EntityTooLarge(Json.toJson(BaseError.entityTooLargeError(s"Your message size must be less than $limit bytes"))))
      case Failure(_)               => Left(BadRequest(Json.toJson(BaseError.badRequestError("Invalid content-length value"))))
    }`